dotCMS/src/main/java/com/dotcms/ai/util/AIUtil.java

```diff
@@ -0,0 +1,68 @@
+package com.dotcms.ai.util;
+
+import com.dotcms.ai.app.AppKeys;
+import com.dotcms.security.apps.AppSecrets;
+import com.dotcms.security.apps.Secret;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import io.vavr.control.Try;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+public class AIUtil {
+
+    public static Map<String, Secret> getSecrets(Contentlet contentlet) {
+        return getSecrets(contentlet.getHost());
+    }
+
+    public static Map<String, Secret> getSecrets(String hostId) {
+        Host host = Try.of(() -> APILocator.getHostAPI().find(hostId, APILocator.systemUser(), true)).getOrNull();
```

Comment:
I think secrets that contains private information should be destroyed at least at the end of the request